wallet/core/src/message.rs

```diff
@@ -0,0 +1,265 @@
+use kaspa_hashes::{Hash, PersonalMessageSigningHash};
+use secp256k1::{Error, XOnlyPublicKey};
+
+#[derive(Clone)]
+pub struct PersonalMessage<'a>(&'a str);
+
+impl AsRef<[u8]> for PersonalMessage<'_> {
+    fn as_ref(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+}
+
+pub fn sign_message(msg: PersonalMessage, privkey: &[u8; 32]) -> Result<Vec<u8>, Error> {
+    let hash = calc_personal_message_hash(msg);
+
+    let msg = secp256k1::Message::from_slice(hash.as_bytes().as_slice())?;
+    let schnorr_key = secp256k1::KeyPair::from_seckey_slice(secp256k1::SECP256K1, privkey)?;
+    let sig: [u8; 64] = *schnorr_key.sign_schnorr(msg).as_ref();
+
+    Ok(sig.to_vec())
+}
+
+pub fn sign_message_with_aux_rand(msg: PersonalMessage, privkey: &[u8; 32], aux_rand: &[u8; 32]) -> Result<Vec<u8>, Error> {
+    let hash = calc_personal_message_hash(msg);
+
+    let msg = secp256k1::Message::from_slice(hash.as_bytes().as_slice())?;
+    let schnorr_key = secp256k1::KeyPair::from_seckey_slice(secp256k1::SECP256K1, privkey)?;
+    let curve = secp256k1::Secp256k1::new();
+    let sig: [u8; 64] = *curve.sign_schnorr_with_aux_rand(&msg, &schnorr_key, aux_rand).as_ref();
+
+    Ok(sig.to_vec())
+}
+
+pub fn verify_message(msg: PersonalMessage, signature: Vec<u8>, pubkey: XOnlyPublicKey) -> Result<bool, Error> {
```

Comment:
You should pass a reference for pubkey `&XOnlyPublicKey` here - because to use this function, you have to clone the pubkey, just for it to be used later as a reference in `sig.verify()`...  (You don't feel it in the unit tests because the pubkey is local).

Reply:
Whst does it mean if it returns Ok(false)?

Reply:
@biryukovmaxim it can't return `Ok(false)`. The underlying schnorr verify will return an `Err` if the signature is anything other than a legit one.

I think the correct return signature of `verify_message` should be changed to just bool, since the only error that can be thrown in the underlying command happens when the signature is not legit

Reply:
I'll just go with that